idea/src/org/jetbrains/kotlin/idea/inspections/UnusedSymbolInspection.kt

```diff
@@ -270,6 +270,8 @@ class UnusedSymbolInspection : AbstractKotlinInspection() {
             return false
         }
 
+        if (!ReferencesSearch.search(declaration, useScope).forEach(::checkReference)) return true
```

Comment:
It's Ok but I don't like that we now invoke references search twice, it can affect performance. I'd consider here to do this only for particular case with `JvmName`. Will fix myself.